diff -ruN ./openapi-cpp-qt-client.origin/CMakeConfig.mustache ./openapi-cpp-qt-client/CMakeConfig.mustache
--- ./openapi-cpp-qt-client.origin/CMakeConfig.mustache	2023-11-29 00:53:49.779576084 +0800
+++ ./openapi-cpp-qt-client/CMakeConfig.mustache	2023-11-29 00:58:21.745804685 +0800
@@ -1,5 +0,0 @@
-@PACKAGE_INIT@
-
-include(${CMAKE_CURRENT_LIST_DIR}/@PROJECT_NAME@Targets.cmake)
-
-check_required_components("@PROJECT_NAME@")
diff -ruN ./openapi-cpp-qt-client.origin/CMakeLists.txt.mustache ./openapi-cpp-qt-client/CMakeLists.txt.mustache
--- ./openapi-cpp-qt-client.origin/CMakeLists.txt.mustache	2023-11-29 00:53:49.779576084 +0800
+++ ./openapi-cpp-qt-client/CMakeLists.txt.mustache	2023-11-29 00:57:52.529002968 +0800
@@ -61,32 +61,3 @@
     PRIVATE
         ${ZLIB_LIBRARIES}{{/contentCompression}}
 )
-
-configure_package_config_file(${CMAKE_CURRENT_SOURCE_DIR}/Config.cmake.in
-    "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
-    INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
-)
-
-install(
-    TARGETS ${PROJECT_NAME}
-    EXPORT ${PROJECT_NAME}Targets
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
-    INCLUDES DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}"
-)
-
-install(
-    FILES ${HEADER_FILES}
-    DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/${PROJECT_NAME}
-)
-
-install(
-    FILES "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake"
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
-)
-
-install(
-    EXPORT ${PROJECT_NAME}Targets
-    DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
-)
diff -ruN ./openapi-cpp-qt-client.origin/Project.mustache ./openapi-cpp-qt-client/Project.mustache
--- ./openapi-cpp-qt-client.origin/Project.mustache	2023-11-29 00:53:49.779576084 +0800
+++ ./openapi-cpp-qt-client/Project.mustache	2023-11-29 00:58:03.127842742 +0800
@@ -1,47 +0,0 @@
-QT += network
-
-HEADERS += \
-# Models
-{{#models}}
-{{#model}}
-    $${PWD}/{{classname}}.h \
-{{/model}}
-{{/models}}
-# APIs
-{{#apiInfo}}
-{{#apis}}
-{{#operations}}
-    $${PWD}/{{classname}}.h \
-{{/operations}}
-{{/apis}}
-{{/apiInfo}}
-# Others
-    $${PWD}/{{prefix}}Helpers.h \
-    $${PWD}/{{prefix}}HttpRequest.h \
-    $${PWD}/{{prefix}}Object.h \
-    $${PWD}/{{prefix}}Enum.h \
-    $${PWD}/{{prefix}}HttpFileElement.h \
-    $${PWD}/{{prefix}}ServerConfiguration.h \
-    $${PWD}/{{prefix}}ServerVariable.h \
-    $${PWD}/{{prefix}}Oauth.h
-
-SOURCES += \
-# Models
-{{#models}}
-{{#model}}
-    $${PWD}/{{classname}}.cpp \
-{{/model}}
-{{/models}}
-# APIs
-{{#apiInfo}}
-{{#apis}}
-{{#operations}}
-    $${PWD}/{{classname}}.cpp \
-{{/operations}}
-{{/apis}}
-{{/apiInfo}}
-# Others
-    $${PWD}/{{prefix}}Helpers.cpp \
-    $${PWD}/{{prefix}}HttpRequest.cpp \
-    $${PWD}/{{prefix}}HttpFileElement.cpp \
-    $${PWD}/{{prefix}}Oauth.cpp
